runtime/dancebox/src/migrations.rs

```diff
@@ -209,19 +210,63 @@ where
     }
 }
 
+pub struct MigrateConfigurationFullRotationPeriod<T>(pub PhantomData<T>);
+impl<T> Migration for MigrateConfigurationFullRotationPeriod<T>
+where
+    T: pallet_configuration::Config,
+{
+    fn friendly_name(&self) -> &str {
+        "TM_MigrateConfigurationFullRotationPeriod"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        log::info!(target: LOG_TARGET, "migrate");
+
+        Configuration::set_full_rotation_period(RuntimeOrigin::root(), 24u32)
```

Comment:
I dont think this is going to work. set full rotation period schedules the configuration change, but wont be applied until after 2 sessions. During those two sessions the configuration wont be readible in the system, as the metadata is not the same

Reply:
we need to read the existing values in currentConfig and pendingConfigs, and modify so that it accepts the new structure

Reply:
There are some examples in https://github.com/paritytech/polkadot-sdk/commit/3bbb336ea707ee1149a6ff689d3a237138f18b2c#diff-39d3b12d624dbf61df8f5ea6fc18a9f12611347f4d730388d6a96deafbba5a71, although I am not sure they are super clear. I think its better to just read whatever value it exists there with the previous struct format, and you just add the new parameter